src/Text.tsx

```diff
@@ -1,13 +1,28 @@
+import React from 'react'
 import styled from 'styled-components'
 import {COMMON, SystemCommonProps, SystemTypographyProps, TYPOGRAPHY} from './constants'
 import sx, {SxProp} from './sx'
-import {ComponentProps} from './utils/types'
 
-const Text = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
+const RestrictedAs = {
+  inlineText: ['span', 'p', 'b', 'i', 'em', 'small', 'strong', 'u', 'code', 'q', 's', 'label', 'div'],
+} as const
```

Comment:
This is mainly just as an example, since the structure might change if we did keep this solution. 

I wasn't too sure where/if we keep constants somewhere specific? I saw our `constants.ts` file, but wasn't sure if this was the proper place to put variables like this?